pandas/core/internals/managers.py

```diff
@@ -1634,21 +1634,17 @@ def fast_xs(self, loc):
 
 
 def create_block_manager_from_blocks(blocks, axes: List[Index]) -> BlockManager:
-    try:
-        if len(blocks) == 1 and not isinstance(blocks[0], Block):
-            # if blocks[0] is of length 0, return empty blocks
-            if not len(blocks[0]):
-                blocks = []
-            else:
-                # It's OK if a single block is passed as values, its placement
-                # is basically "all items", but if there're many, don't bother
-                # converting, it's an error anyway.
-                blocks = [
-                    make_block(
-                        values=blocks[0], placement=slice(0, len(axes[0])), ndim=2
-                    )
-                ]
+    if len(blocks) == 1 and not isinstance(blocks[0], Block):
```

Comment:
didn't you have a suggestion to simplify this? 
wonder where we actually hit this (the no block case)

Reply:
Whoops, the test got lost as I was fiddling with local branches. Gimme a sec.

I saw that `create_block_manager_from_blocks` was left in the API for backward compatibility, I didn't want to fiddle with it.

Reply:
if you can remove / simplify happy to have that